sonar-dotnet-shared-library/src/main/java/org/sonar/plugins/dotnet/tests/DotCoverReportParser.java

```diff
@@ -100,13 +105,34 @@ private void collectCoverage(String fileCanonicalPath, String contents) {
 
       matcher = COVERED_LINES_PATTERN_2.matcher(highlightedContents);
 
+      Set<SequencePoint> sequencePoints = new HashSet<>();
+
       while (matcher.find()) {
-        int line = Integer.parseInt(matcher.group(1));
-        int hits = Integer.parseInt(matcher.group(2));
-        coverage.addHits(fileCanonicalPath, line, hits);
+        int lineStart = Integer.parseInt(matcher.group(1));
+        int lineEnd = Integer.parseInt(matcher.group(2));
+        int hits = Integer.parseInt(matcher.group(3));
+
+        coverage.addHits(fileCanonicalPath, lineStart, hits);
+
+        if (lineStart == lineEnd) {
+          sequencePoints.add(new SequencePoint(lineStart, hits));
+        }
 
         LOG.trace("dotCover parser: found coverage for line '{}', hits '{}' when analyzing the path '{}'.",
-            line, hits, fileCanonicalPath);
+            lineStart, hits, fileCanonicalPath);
+      }
+
+      Map<Integer, List<SequencePoint>> collect = sequencePoints.stream().collect(Collectors.groupingBy(SequencePoint::getStartLine));
```

Comment:
s/collect/sequencePointsPerLine ?

Reply:
Yes. I thought to change it and then forgot 😄